eduid_am/consistency_checks.py

```diff
@@ -152,6 +162,62 @@ def unverify_nins(userdb, user_id, nins):
                     count += 1
                     logger.debug('Old user NINs AFTER: {}.'.format(user.nins.to_list()))
                     userdb.save(user)
-    except DocumentDoesNotExist:
-        pass
+        except DocumentDoesNotExist:
+            pass
     return count
+
+
+def check_locked_identity(userdb, user_id, attributes, app_name):
+    """
+    :param userdb: Central userdb
+    :param user_id: User document _id
+    :param attributes: attributes to update
+    :param app_name: calling application name, like 'eduid_signup'
+
+    :type userdb: eduid_userdb.userdb.UserDB
+    :type user_id: bson.ObjectId
+    :type attributes: dict
+    :type app_name: string
+
+    :return: attributes to update
+    :rtype: dict
+    """
+    # Check verified nins that will be set against the locked_identity attribute,
+    # if that does not exist it should be created
+    set_attributes = attributes.get('$set', {})
+    nins = set_attributes.get('nins', [])
+    verified_nins = [nin for nin in nins if nin.get('verified')]
+    if not verified_nins:
+        return attributes  # No verified nins will be set
+
+    # A user can not have more than one verified nin at this time
+    if len(verified_nins) != 1:
+        logger.error('Tried to set more than one verified nin for user with id {}'.format(user_id))
+        raise EduIDUserDBError('Tried to set more than one verified nin for user.')
+
```

Comment:
I would set

  nin = verified_nins[0]

here instead of using the verified_nins[0] below. Clearer intent and more elegant code.